tests/test_jsonpointer.py

```diff
@@ -75,30 +119,42 @@ def test_extend_jsonpointer_multi_keys(value, newkeys):
     assert str(pointer) == value + ''.join(f'/{jsonpointer_escape(key)}' for key in newkeys)
 
 
+def test_uri_fragment_safe_characters():
+    pointer_str = "/!$&'()*+,;="
+    pointer = JSONPointer(pointer_str)
+    assert pointer.uri_fragment() == pointer_str
+
+
+@pytest.mark.parametrize('jp_cls', (JSONPointer, JPtr))
 @given(json, jsonpointer_key)
-def test_evaluate_jsonpointer(value, testkey):
-    assert JSONPointer().evaluate(value) == value
-    assert JSONPointer().evaluate(JSON(value)) == value
+def test_evaluate_jsonpointer(jp_cls, value, testkey):
+    assert jp_cls().evaluate(value) == value
+    assert jp_cls().evaluate(JSON(value)) == value
 
     generate_jsonpointers(resolved_pointers := {}, value)
     for pointer, target in resolved_pointers.items():
-        assert JSONPointer(pointer).evaluate(value) == target
-        assert JSONPointer(pointer).evaluate(JSON(value)) == target
+        assert jp_cls(pointer).evaluate(value) == target
+        assert jp_cls(pointer).evaluate(JSON(value)) == target
 
     if isinstance(value, list):
-        with pytest.raises(JSONPointerReferenceError):
-            JSONPointer(f'/{len(value)}').evaluate(value)
-        with pytest.raises(JSONPointerReferenceError):
-            JSONPointer('/-').evaluate(value)
-        with pytest.raises(JSONPointerReferenceError):
-            JSONPointer('/').evaluate(value)
+        with pytest.raises(jp_cls.reference_exc) as exc_info:
+            jp_cls(f'/{len(value)}').evaluate(value)
+            assert exc_info.type == jp_cls.reference_exc
```

Comment:
This assertion is not being hit due to the exception; it needs to be outside the `with` block.

Reply:
Good catch- I could have sworn I put all of these outside (I figured it out partway through).  Looks like I missed about half of them 😬